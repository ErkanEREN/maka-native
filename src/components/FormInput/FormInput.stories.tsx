import React from 'react'
import { View } from 'react-native'
import { storiesOf } from '@storybook/react-native'

import FormInput from './FormInput'

storiesOf('Form Input', module).add('FormInput', () => (
  <View>
    <FormInput
      name="test"
      placeholder="Test"
      onChangeText={value => console.log(value)}
    />
  </View>
))
