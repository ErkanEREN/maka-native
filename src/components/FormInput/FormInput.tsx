import React from "react";
import { StyleSheet, TextInput } from "react-native";

type Props = {
  error?: string;
  value?: string;
  onChangeText: (name: string, value: string) => void;
  style?: {};
  name: string;
  placeholder: string;
};

export default function FormInput({
  style,
  error,
  value,
  onChangeText,
  name,
  placeholder
}: Props) {
  return (
    <TextInput
      style={[styles.textInput, style]}
      value={value}
      placeholder={placeholder}
      onChangeText={(value: string) => onChangeText(name, value)}
    />
  );
}

const styles = StyleSheet.create({
  textInput: {
    height: 40,
    borderBottomWidth: StyleSheet.hairlineWidth,
    marginBottom: 20
  }
});