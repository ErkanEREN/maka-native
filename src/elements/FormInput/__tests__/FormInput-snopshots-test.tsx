import React from "react";
import renderer from "react-test-renderer";

import FormInput from "../FormInput";

describe("FormInput", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const placeholder = "form input placeholder";
      const onChangeText = jest.fn();
      const name = "textinput-name";
      const tree = renderer
        .create(
          <FormInput
            placeholder={placeholder}
            onChangeText={onChangeText}
            name={name}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
