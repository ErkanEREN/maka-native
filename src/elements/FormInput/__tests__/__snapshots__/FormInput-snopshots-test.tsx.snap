// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Rendering should match to snapshot 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    className="sc-bdVaJa hAFfYc"
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="form input placeholder"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
  />
</View>
`;
